src/main/java/uk/gov/hmcts/cmc/claimstore/services/OffersService.java

```diff
@@ -3,69 +3,74 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.cmc.ccd.domain.CaseEvent;
 import uk.gov.hmcts.cmc.claimstore.events.EventProducer;
 import uk.gov.hmcts.cmc.claimstore.exceptions.ConflictException;
-import uk.gov.hmcts.cmc.claimstore.processors.JsonMapper;
-import uk.gov.hmcts.cmc.claimstore.repositories.OffersRepository;
+import uk.gov.hmcts.cmc.claimstore.services.search.CaseRepository;
 import uk.gov.hmcts.cmc.domain.models.Claim;
 import uk.gov.hmcts.cmc.domain.models.offers.MadeBy;
 import uk.gov.hmcts.cmc.domain.models.offers.Offer;
 import uk.gov.hmcts.cmc.domain.models.offers.Settlement;
 
-import java.time.LocalDateTime;
-
 import static java.lang.String.format;
 
 @Service
 public class OffersService {
 
     private final ClaimService claimService;
-    private final OffersRepository offersRepository;
-    private final JsonMapper jsonMapper;
+    private final CaseRepository caseRepository;
     private final EventProducer eventProducer;
+    private final TimeService timeService;
 
     @Autowired
     public OffersService(
         ClaimService claimService,
-        OffersRepository offersRepository,
+        CaseRepository caseRepository,
         EventProducer eventProducer,
-        JsonMapper jsonMapper
+        TimeService timeService
     ) {
         this.claimService = claimService;
-        this.offersRepository = offersRepository;
+        this.caseRepository = caseRepository;
         this.eventProducer = eventProducer;
-        this.jsonMapper = jsonMapper;
+        this.timeService = timeService;
     }
 
-    public void makeOffer(Claim claim, Offer offer, MadeBy party) {
+    public void makeOffer(Claim claim, Offer offer, MadeBy party, String authorisation) {
         assertSettlementIsNotReached(claim);
 
         Settlement settlement = claim.getSettlement().orElse(new Settlement());
         settlement.makeOffer(offer, party);
-        offersRepository.updateSettlement(claim.getId(), jsonMapper.toJson(settlement));
+
+        caseRepository.updateSettlement(claim, settlement, authorisation,
+            eventName("OFFER_MADE_BY", party.name()), null);
```

Comment:
I'm not that happy with having to pass `null` value around. Can we please split it to 2 methods like it's done with the database, e.g.
- `updateSettlementNegotiation` which updates with a new offer or rejection,
- `reachSettlementAgreement` where one of the parties accepts the other's offer and the settlement date is recorded.